### ISSUE TITLE
Tests fail at high build concurrency for default openssh config

### ISSUE THREAD

**Issue by FauxFaux** [alexcrichton/ssh2-rs#135]
`cargo test` fails for me most of the time with `--test-threads=24`, even on a dual core machine, and ~30% of the time with `$(nproc)` == 12.

Random tests fail with: `'called Result::unwrap() on an Err value: Error { code: -43, msg: "Failed getting banner" }'`, at https://github.com/alexcrichton/ssh2-rs/blob/cab5f0fc9f217a577ad7614e5539e0d0315279b1/tests/all/main.rs#L27
`-43` is `LIBSSH2_ERROR_SOCKET_RECV`.

If you enable debugging (`cfg.define("LIBSSH2DEBUG", None); cfg.define("HAVE_GETTIMEOFDAY", None);` and [cripple the filtering](https://github.com/wez/libssh2/blob/85ebd53f59def688703497ebe63dcf489b9aae71/src/misc.c#L449)) you can see that the connection is being closed.

Initially I thought this was a horrible thread safety bug (and spent ages on that), but it's probably just openssh's default `MaxSessions 10`, which means no more than 10 people can be logging in at a time, so, uh, probably a minor documentation note, instead of the cool bug I was hoping for?

**Comment by wez** [comment]
yeah, that's frustrating.  I spent some time troubleshooting failures in the CI and cooked up this script.  What seemed to be the critical thing there was the MaxStartups value:

https://github.com/alexcrichton/ssh2-rs/blob/master/tests/run_integration_tests.sh#L51

In terms of making things more discoverable for `cargo test`, what do you think about having `run_integration_tests.sh` export an env var to indicate that it is being used to run the tests, and then having the tests check for that and emitting a message to tell the user to run the script instead of `cargo test` directly?

**Comment by FauxFaux** [comment]
That's a nice idea.

Much better than my proposal (documentation on github!), or trying to teach the tests to limit concurrency themselves.
